server/src/main/java/org/elasticsearch/index/mapper/GeoShapeIndexer.java

```diff
@@ -125,7 +127,7 @@ public Void visit(Point point) {
 
         @Override
         public Void visit(Polygon polygon) {
-            addFields(LatLonShape.createIndexableFields(name, GeoShapeUtils.toLucenePolygon(polygon)));
+            addFields(LatLonShape.createIndexableFields(name, GeoShapeUtils.toLucenePolygon(polygon), true));
```

Comment:
The additional checks in the tesselator also add some performance cost.

Reply:
Yes, the hope is that the overhead of checking for intersection in the tessellate is compensated with skipping the normalisation step.